src/cli/telemetry/telemetry.ts

```diff
@@ -0,0 +1,304 @@
+
+  const details = sys.details;
+  const versions = getCoreCompiler()?.versions || { typescript: 'unknown', rollup: 'unknown' };
+  let duration = undefined;
+  let error: any;
+
+  if (action) {
+    const start = new Date();
+
+    try {
+      await action();
+    } catch (e) {
+      error = e;
+    }
+
+    const end = new Date();
+    duration = end.getTime() - start.getTime();
+  }
+
+  // We'll get componentCount details inside the taskBuild, so let's not send two messages.
+  if (!tracking || (flags.task == 'build' && !flags.args.includes('--watch'))) {
+    return;
+  }
+
+  const session_id = await getTelemetryToken();
+  const packages = await getInstalledPackages();
+  const targets = await getActiveTargets(validatedConfig.config);
+
+  const data: TrackableData = {
+    session_id,
+    yarn: isUsingYarn(),
+    duration_ms: duration,
+    targets,
+    packages: packages || [],
+    arguments: flags.args,
+    task: flags.task,
+    stencil: getCoreCompiler()?.version || 'unknown',
+    system: `${sys.name} ${sys.version}`,
+    os_name: details.platform,
+    os_version: details.release,
+    cpu_model: `${details.cpuModel}`,
+    build: getCoreCompiler()?.buildId || 'unknown',
+    typescript: versions.typescript,
+    rollup: versions.rollup,
+  };
+
+  await sendMetric('stencil_cli_command', data);
+  logger.debug(`${logger.blue('Telemetry')}: ${logger.gray(JSON.stringify(data))}`);
+
+  if (error) {
+    throw error;
+  }
+}
+
+function isUsingYarn() {
+  return getCompilerSystem().getEnvironmentVar('npm_execpath').includes('yarn');
+}
+
+async function getActiveTargets(config: Config): Promise<string[][] | OutputTargetOptions[][]> {
+  return (config.outputTargets as any[]).map(target => {
+    let outputTargetSettings: OutputTargetOptions = {};
+
+    if (target.type === 'www') {
+      outputTargetSettings.serviceWorker = isObject(target?.serviceWorker);
+    }
+
+    return [target.type, outputTargetSettings];
+  });
+}
+
+/**
+ * Reads package-lock.json and package.json files in order to cross references the dependencies and devDependencies properties. Pull the current installed version of each package under the @stencil, @ionic, and @capacitor scopes.
+ * @returns string[]
+ */
+async function getInstalledPackages() {
+  try {
+    // Read package.json and package-lock.json
+    const appRootDir = getCompilerSystem().getCurrentDirectory();
+
+    const packageJson: PackageJsonData = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package.json'),
+    );
+
+    const packageLockJson: any = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package-lock.json'),
+    );
+
+    // They don't have a package.json for some reason? Eject button.
+    if (!!packageJson) {
+      return [];
+    }
+
+    const packages = Object.entries({
+      ...packageJson.devDependencies,
+      ...packageJson.dependencies,
+    }) as [string, string][];
```

Comment:
I don't know if we need this type assertion, as this call to `Object.entries` should give us a `[string, string][]` already (if you want to be declarative, there's nothing wrong with putting the type directly following `packages` like `const packages: [string, string][] = ...`)